vignettes/rhino-fluent.Rmd

```diff
@@ -0,0 +1,712 @@
+ui <- function(id) {
+  ns <- NS(id)
+  div(
+    Text("A randomly generated dataset of imaginary sales deals", variant = "large"),
+    div(
+      class = "ms-depth-4",
+      Toggle.shinyInput(ns("includeOpen"), label = "Include open deals")
+    ),
+    Stack(
+      horizontal = TRUE,
+      tokens = list(padding = 20, childrenGap = 10),
+      span(class = "ms-depth-8", uiOutput(ns("datatable"))),
+      span(class = "ms-depth-8", plotlyOutput(ns("barchart")))
+    )
+  )
+}
+
+#' @export
+server <- function(id) {
+  moduleServer(id, function(input, output, session) {
+    filtered_deals <- reactive({
+      shiny.fluent::fluentSalesDeals |> filter(
+        is_closed | input$includeOpen
+      )
+    })
+    output$datatable <- renderUI({
+      DetailsList(items = filtered_deals(), columns = columns)
+    })
+    output$barchart <- renderPlotly({
+      generate_barchart(filtered_deals())
+    })
+  })
+}
+```
+
+Here, we have called the `plotly` library, and also the function that we defined in the `app/logic` folder. `plotlyOutput` will help us to render the graph.
+
+Similar to previous usecase, we have added `plotlyOutput` inside the `Stack` so that it is rendered horizontally beside the table as a card. It has the same styling as of the data table's card, hence no need to define the style separately. 
+
+In the server section, we use `renderPlotly` to ask the backend of the application for rendering the barchart.
+
+Once ready, on running the application, we should see the plotly bar chart rendering successfully and also responding to our interactions with the toggle input button.
+
+```{r, echo=FALSE}
+knitr::include_graphics("images/rhino_fluent_gif_3.gif")
+```
+
+## Create rhino modules for toggle button and barchart
+
+Now that we have successfully developed a working application with a data table, a toggle input and a barchart, let us further create different rhino modules for these features!
+
+We already have the datatable module with us. We will thus create two more modules - `app/view/toggle.R` and `app/view/plot.R` along with some modifications to `app/view/datatable`. 
+
+These modules will br called from the main file.
+
+### The `app/view/toggle.R` module -
+
+```
+# app/view/toggle.R
+
+box::use(
+  dplyr[filter],
+  shiny[div, moduleServer, NS, observeEvent, reactive],
+  shiny.fluent[Text, Toggle.shinyInput],
+)
+
+#' @export
+ui <- function(id) {
+  ns <- NS(id)
+  div(
+    Text("A randomly generated dataset of imaginary sales deals", variant = "large"),
+    div(
+      class = "ms-depth-4",
+      Toggle.shinyInput(ns("includeOpen"), label = "Include open deals")
+    )
+  )
+}
+
+#' @export
+server <- function(id) {
+  moduleServer(id, function(input, output, session) {
+    filtered_deals <- reactive({
+      shiny.fluent::fluentSalesDeals |> filter(
+        is_closed | input$includeOpen
+      )
+    })
+    observeEvent(input$includeOpen, {
+      filtered_deals()
+    })
+    return(filtered_deals())      
```

Comment:
This is not correct. `filtered_deals` reactive includes `input$includeOpen` inside, thus it will be recalculated every time it is changed. `observeEvent` is redundant. Also, the module should return `filtered_deals`, not `filtered_deals()` - you want to return a reactive object, not a value.